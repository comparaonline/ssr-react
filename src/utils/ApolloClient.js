import fetch from 'isomorphic-fetch';
import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { createHttpLink } from 'apollo-link-http';
import { get } from 'Config';

if (!global.fetch) {
  global.fetch = fetch;
}

export const createApolloClientSSR = () => (
  new ApolloClient({
    ssrMode: true,
    link: createHttpLink({
      uri: get('graphqlEndpoint'),
    }),
    cache: new InMemoryCache(),
  })
);

export const createApolloClient = state => (
  new ApolloClient({
    ssrMode: false,
    link: createHttpLink({
      uri: get('graphqlEndpoint'),
    }),
    cache: new InMemoryCache().restore(state),
  })
);
