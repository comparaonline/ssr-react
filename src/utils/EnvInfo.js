const env = process.env.NODE_ENV;
const analyze = process.env.ANALYZE;

const devEnvs = [
  'development',
];

const prodEnvs = [
  'production',
  'staging',
  'qc',
];

const isDevEnv = devEnvs.includes(env);
const isProdEnv = prodEnvs.includes(env);

const isServer = typeof window === 'undefined';
const isClient = !isServer;

export { env, isDevEnv, isProdEnv, isClient, isServer, analyze };
