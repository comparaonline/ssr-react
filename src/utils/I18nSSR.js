export const getInitialLanguage = req => req.i18n.language;

export const getInitialState = (req) => {
  const {
    languages,
    services: {
      resourceStore: { data },
    },
  } = req.i18n;

  const initialState = languages.reduce((acc, lang) => {
    acc[lang] = data[lang];
    return acc;
  }, {});

  return initialState;
};
