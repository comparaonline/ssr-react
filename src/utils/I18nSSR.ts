import { Request } from 'Types/express';
import { i18n, Lang } from 'Types/i18n';
import { Ii18n } from '../server/middlewares/i18next';

export const getInitialLanguage = (req: Request): Ii18n['language'] => req.i18n.language;

export const getInitialState = (req: Request): Partial<i18n> => {
  const {
    languages,
    services: {
      resourceStore: { data },
    },
  } = req.i18n;

  const initialState: Partial<i18n> = languages.reduce((acc: Partial<i18n>, lang: Lang) => {
    acc[lang] = data[lang];
    return acc;
  }, {});

  return initialState;
};
