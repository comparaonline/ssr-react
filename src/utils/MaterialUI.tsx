import React from 'react';
import { Theme,  createMuiTheme } from '@material-ui/core/styles';
import blue from '@material-ui/core/colors/blue';

export const jssID: string = 'jss-server-side';

export const createTheme = (theme?: Theme) => createMuiTheme({
  typography: {
    useNextVariants: true,
  },
  palette: {
    primary: blue,
    type: 'light'
  },
  ...theme,
});

export class RemoveStylesOnClient extends React.Component {
  componentDidMount() {
    const jssStyles: HTMLElement = document.getElementById(jssID);

    if (jssStyles && jssStyles.parentNode) {
      jssStyles.parentNode.removeChild(jssStyles);
    }
  }

  render() {
    return this.props.children;
  }
}
