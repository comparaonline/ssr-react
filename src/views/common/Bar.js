import React, { Component } from 'react';

class Bar extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount() {
    console.log('Bar Component Mounted!');
  }

  render() {
    return (
      <div>
        <h2>Bar component loaded</h2>
        <span>this is an async component</span>
      </div>
    );
  }
}

export default Bar;
