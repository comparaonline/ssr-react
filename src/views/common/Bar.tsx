import React, { Component } from 'react';
import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import { connect } from 'react-redux';
import { changeSSR } from 'Redux/actions';
import { State } from 'Redux/reducers';

import { Title } from './Layout';

import * as styles from './styles.css';

interface IProps {
  isSSR: State['isSSR'];
  changeSSR: (ssr: boolean) => void;
};

export class Bar extends Component<IProps, {}> {
  constructor(props: IProps) {
    super(props);
    this.state = {};
  }

  componentDidMount() {
    console.log('Bar Component Mounted!'); // eslint-disable-line
  }

  onClick = () => {
    this.props.changeSSR(!this.props.isSSR);
  }

  render() {
    return (
      <div className={styles.tanBG}>
        <Grid container spacing={16} justify="center">
          <Grid item>
            <Title>Bar component loaded</Title>
            <span>this is an async component</span>
            <div>isSSR: {this.props.isSSR.toString()}</div>
            <Grid container spacing={16} justify="center">
              <Grid item>
                <Button variant="contained" color="secondary" onClick={this.onClick} className={styles.changeStateButton}>change state</Button>
              </Grid>
            </Grid>
          </Grid>
        </Grid>
      </div>
    );
  }
}

const mapStateToProps = (state: State) => ({
  isSSR: state.isSSR,
});

export default connect(mapStateToProps, { changeSSR })(Bar);
