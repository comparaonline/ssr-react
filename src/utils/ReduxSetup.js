import { createStore, compose } from 'redux';
import { isDevEnv } from 'Utils/EnvInfo';
import rootReducer from 'Redux/reducers';

export const configureStore = (initialState = {}) => {
  const composeEnhancers = isDevEnv
    ? (typeof window !== 'undefined' &&
        window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) ||
      compose
    : compose;

  const store = createStore(
    rootReducer,
    initialState,
    composeEnhancers(),
  );

  return store;
};

export const getInitialState = (opts = {}) => {
  const { req } = opts; // eslint-disable-line

  return {
    isSSR: true,
  };
};
