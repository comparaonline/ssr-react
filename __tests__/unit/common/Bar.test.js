import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Bar } from 'Views/common/Bar';

import { renderComponent } from '../../utils';

describe('Bar Component', () => {
  let wrapper = null;
  let changeSSR = null;

  beforeEach(() => {
    changeSSR = jest.fn();
    wrapper = shallow(renderComponent(Bar, {
     isSSR: true,
     changeSSR,
    }));
  });

  test('it should match with snapshot', () => {
    const tree = renderer.create(renderComponent(Bar, {
      isSSR: true,
      changeSSR: jest.fn(),
    }));

    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('click button should call onClick function', () => {
    wrapper.find('button').simulate('click');
    expect(changeSSR.mock.calls.length).toBe(1);
  });
});
