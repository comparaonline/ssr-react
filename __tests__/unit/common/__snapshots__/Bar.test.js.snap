// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar Component it should match with snapshot 1`] = `
<div>
  <h2
    className="sc-bdVaJa rOCEJ"
  >
    Bar component loaded
  </h2>
  <span>
    this is an async component
  </span>
  <div>
    isSSR: 
    true
  </div>
  <button
    onClick={[Function]}
  >
    change state
  </button>
</div>
`;
